### DISCUSSION TITLE
Would you provide a Type of Tenant<TKey>, I want to custom my primary key type.

### DISCUSSION THREAD

**Discussion by codelovercc** [Finbuckle/Finbuckle.MultiTenant#568] [Ideas]
Hi,guys.
I don't like primary key as a string.Would you provide a Type of Tenant<TKey>? I want to custom my primary key type,just like Identity core's IdentityUser<TKey>.
Thank you very much.

**Comment by AndrewTriesToCode** [comment]
Hi, yeah I'm working on this. It is unfortunately pretty complicated but should be doable.

**Reply by ThomasVague** [reply]
Sort of related question; any plans on making the name of the TenantId configurable? I already have a project I want to implement Finbuckle with, but our key is named CustomerId.

**Comment by vigouredelaruse** [comment]
further to this, please consider the following snapshot of redis cache keyspace when useing idistributedcachestore. note the 'naked' keynames as opposed to the recommended keyspace as per https://www.w3schools.io/nosql/redis-keys-naming-convention/

![image](https://user-images.githubusercontent.com/31052867/185680402-16ad4b7b-5b8f-4410-bf4c-449d28591fb4.png)

i want to combine the following improvement requests
```
- Tenant<TKey>
- Tenanant<TKey, TCacheOptions>, where TCacheOptions exposes CacheEntryName + CacheEntryExpirationTime
```
the use case is prevention of keyspace collisions in situations of deliberate or accidental sharing of redis instances

so to be clear this is an improvement request that would 
- expose cache key names as runtime configurable options, the better to support roll your own key name generation (obfuscation/hashing) 
- ~~expose cache key expiration~~ sorry already supported
- for extra points, detect namespace collisions when 2 different sites (so not instances in a load balanced kubernetes pod let's say, but 2 different developers pointing at the same redis for whatever reason ) use the same cache

best regards
